src/snippet.rs

```diff
@@ -44,6 +48,14 @@ impl<'a> Group<'a> {
         self
     }
 
+    /// Set the primary [`Level`] for this [`Group`]. If not specified, the
+    /// [`Level`] of the first [`Title`] in the [`Group`] will be used, if no
+    /// [`Title`] is present, it will default to [`Level::ERROR`].
+    pub fn primary_level(mut self, level: Level<'a>) -> Self {
```

Comment:
a bit hesitant with the name but we can worry about that in a more holistic review of the API rather than the pigeonhole of this specific PR